app/forms/waste_carriers_engine/contact_email_form.rb

```diff
@@ -4,10 +4,13 @@ module WasteCarriersEngine
   class ContactEmailForm < BaseForm
     attr_accessor :contact_email, :confirmed_email
 
+    validates :contact_email, :confirmed_email, "defra_ruby/validators/email": true
+    validates :confirmed_email, "waste_carriers_engine/matching_email": { compare_to: :contact_email }
+
     def initialize(transient_registration)
       super
-      self.contact_email = @transient_registration.contact_email
-      self.confirmed_email = @transient_registration.contact_email
+      self.contact_email = transient_registration.contact_email
```

Comment:
NIT Adding a line between super and this as per other changes

```suggestion
      
      self.contact_email = transient_registration.contact_email
```